### PR TITLE
[FIX] Fix Future Warnings in ivp.py and test_quad.py and RuntimeError in lq_control.py.

### PR DESCRIPTION
**Pull request by duncanhobbs**
This PR fixes some warnings raised in issues \#505 \#506 and \#507.

1) FutureWarning in `ivp.py` (see issue \#505).

2) RuntimeError in `lq_control.py` (see issue \#506)

3) Pandas related future warnings in `test_quad.py` (see issue \#507).

This is my first QuantEcon PR so let me know if I need to fix anything.

### CODE REVIEW

**Comment by mmcky** [comment]
thanks @duncanhobbs -- this will be helpful. I will review this in the next couple of days.

**Comment by duncanhobbs** [comment]
Thanks @oyamad.

I  adjusted the code to incorporate what you suggested, and am going through the other files that use LQ to make sure the other tests do not break. 

In test_lqnash.py [here](https://github.com/QuantEcon/QuantEcon.py/blob/master/quantecon/tests/test_lqnash.py#L32) it raises `ValueError 'beta must be strictly smaller than 1'`.

I think this is because `beta = 1` and `C = None` by default.

Should the `ValueError` be thrown if `C = None`? 

Or only if `C != 0 or C is not None`?

**Comment by oyamad** [comment]
@duncanhobbs Sorry my code was wrong. `C != 0` should have been `self.C != 0`.
And it will be better to provide more information in the error message.

Try this:

```py
if (self.C != 0).any() and beta >= 1:
    raise ValueError('beta must be strictly smaller than 1' +
                     'when T is None and C != 0')
```

**Comment by duncanhobbs** [comment]
Thanks @oyamad

I made the tweak and it solved the issue in `test_lqnash.py`.

When I ran the test suite again, the ValueError was raised in test_robust_lq.py [here](https://github.com/QuantEcon/QuantEcon.py/blob/master/quantecon/tests/test_robustlq.py#L49-L51).

If I assign `beta = beta` and `theta = theta` the tests pass. 

Is that okay to do?

**Comment by oyamad** [comment]
The problem is in this line: https://github.com/QuantEcon/QuantEcon.py/blob/d9c1e8a1664b422edbeed42cd6065968187de739/quantecon/tests/test_robustlq.py#L51

`beta` is an optional argument, so it has to be passed as `beta=beta`:

```py
        self.lq_test = LQ(Q, R, A, B, C, beta=beta)
```

(See also QuantEcon/lecture-source-py#314)

**Comment by duncanhobbs** [comment]
Hi @oyamad 

I have made the changes you requested.

I am not sure why the other commits got added from July 11. It may have happened when I was trying to rebase and clean up the commit history. I forgot to rebase my branch before pushing commits to the pull request.

Would the easiest way to fix this be to rebase the pull request commits or open a new PR with the feature branch updated?

It feels to me that opening a new PR is easier, but I am not sure what standard practice is.

Apologies.

File: quantecon/tests/test_robustlq.py [hunk 1]
```diff
@@ -46,9 +46,9 @@ def setUp(self):
 
         # the *_pf endings refer to an example with pure forecasting
         # (see p171 in Robustness)
-        self.rblq_test = RBLQ(Q, R, A, B, C, beta, theta)
-        self.rblq_test_pf = RBLQ(Q_pf, R, A, B_pf, C, beta, theta)
-        self.lq_test = LQ(Q, R, A, B, C, beta)
+        self.rblq_test = RBLQ(Q, R, A, B, C, beta=beta, theta=theta)
+        self.rblq_test_pf = RBLQ(Q_pf, R, A, B_pf, C, beta=beta, theta=theta)
```
Comment (oyamad) [inline quantecon/tests/test_robustlq.py]: Are the changes in these two lines really necessary?

**Comment by oyamad** [comment]
Thanks @duncanhobbs 
Can you rebase your branch on the current `master` (and then force-push)?

File: quantecon/tests/test_robustlq.py (hunk 1 above)
Comment (duncanhobbs) [reply quantecon/tests/test_robustlq.py]: I see they are not. I will fix this.

**Comment by duncanhobbs** [comment]
Thanks @oyamad.

I rebased and force-pushed as you suggested and things seem to work. 

The tests all pass locally.

Thanks for your patience. I am not very experienced with using git yet.

Let me know if I need to change anything else.

**Comment by jstac** [comment]
Thanks @duncanhobbs, much appreciated.

**Comment by mmcky** [comment]
thanks @duncanhobbs and @oyamad for your reviews and advice on this.

**Pull request merged**